docs: adding ABCI error information (#3072)
cometbft/cometbft: docs/tutorials/go.md

}
```

### 1.3.6 Handling errors

Please note that in the method signature for the ABCI methods, there is a response and an error return, such as
`(*abcitypes.[Method]Response, error)`. Some of the ABCI methods' responses might include a field that can return
an error in the response, such as the `Code` field in the `CheckTxResponse`. The application can use the `Code`

> field to signal CometBFT that the transaction was rejected. Other examples are the `TxResults` in the
> `FinalizeBlockResponse`, the `ExecTxResult` that also has a `Code` field which can be used by the application
> to signal that the transactions didn't execute properly, or the `QueryResponse`. The `QueryResponse` also includes
> a `Code` field to signal that a query to the application was unsuccessful or it could not find the information.

Reviewer: ```suggestion
field to signal CometBFT that the transaction was rejected. Another example is `FinalizeBlockResponse`, which has a `TxResults` array field with each result containing a `Code` field that can be used by the application
to signal that a transaction didn't execute properly. Or `QueryResponse`, which also includes
a `Code` field to signal that a query to the application was unsuccessful or it could not find the information.
```
